### DISCUSSION TITLE
Are there semaphore Variables or flags?

### DISCUSSION THREAD

**Discussion by LeSuedois** [earlephilhower/arduino-pico#1213] [Q&A] [answered]
Hello just some silly question : 
Are there semaphore Variables or Flags or anything like that on this RP2040 core?
I found them useful on the ESP32 when receiving data over USB and stream it to some large Neopixel installs. Sometimes I get glitches when a animation is updating data and the ISR is pushing data to the Neopixels. Quite decent glitches, but glitches anyway.
Thank you,
Simon

**Comment by earlephilhower** [comment, marked as answer]
Yes, we support the Raspberry Pi Pico-SDK native ones.  They're documented here: https://raspberrypi.github.io/pico-sdk-doxygen/group__mutex.html

**Reply by LeSuedois** [reply]
oh cool, I didn't saw that, Thank you very much.
I tried it this morning but no luck, do you know if there is a example anywhere?

**Reply by earlephilhower** [reply]
Anything in the SDK related to mutexes would work.  Something like

````
auto_init_mutex(mx);
void fcn(){ 
   mutex_enter_blocking(&mx);
   ... do dangerous stuff ...
   mutex_exit(&mx);
}
````
